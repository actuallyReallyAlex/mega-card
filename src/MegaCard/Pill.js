import React from 'react'
import { Box, Text } from 'grommet'
import { Info } from 'grommet-icons'
import { buildShadow, colors, defaultProps, propTypes } from '../constants'

const Pill = ({
  background,
  color,
  data,
  indicator,
  shadowSize,
  shadowType,
  title
}) => {
  return (
    <Box
      align="center"
      background={background}
      direction="row"
      gap="medium"
      justify="between"
      margin="small"
      pad="medium"
      round="large"
      style={{ boxShadow: buildShadow(shadowSize, shadowType) }}
    >
      {indicator && (
        <Box>
          <Box
            style={{ backgroundColor: colors[color] }}
            height="8px"
            round="full"
            width="8px"
          />
        </Box>
      )}
      <Box margin={{ right: 'medium' }}>
        <Text size="16px" weight="bold">
          {title}
        </Text>
        <Text size="small" color="dark-4">
          {data}
        </Text>
      </Box>
      <Box>
        <Box
          align="center"
          height="35px"
          justify="center"
          round="full"
          style={{ backgroundColor: colors[color] }}
          width="35px"
        >
          <Info color="white" size="20px" />
        </Box>
      </Box>
    </Box>
  )
}

Pill.propTypes = propTypes

Pill.defaultProps = defaultProps

export default Pill
