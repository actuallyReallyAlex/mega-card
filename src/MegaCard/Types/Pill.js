import React from 'react'
import { Box, Text, Heading } from 'grommet'
import {
  buildShadow,
  colors,
  defaultProps,
  icons,
  propTypes
} from '../../constants'

const Pill = ({
  background,
  color,
  data,
  icon,
  indicator,
  shadowSize,
  shadowType,
  title,
  titleColor = color
}) => {
  const Icon = icons[icon]

  return (
    <Box
      align="center"
      background={background}
      direction="row"
      gap="medium"
      justify="between"
      margin="small"
      pad="medium"
      round="large"
      style={{ boxShadow: buildShadow(shadowSize, shadowType) }}
    >
      {indicator && (
        <Box>
          <Box
            style={{ backgroundColor: colors[color] }}
            height="8px"
            round="full"
            width="8px"
          />
        </Box>
      )}
      <Box align="start" margin={{ right: 'medium' }}>
        <Text size="16px" style={{ color: colors[titleColor] }} weight="bold">
          {title}
        </Text>
        <Heading
          color="dark-2"
          level="5"
          margin={{ bottom: 'none', top: 'xsmall' }}
          style={{ fontSize: '1.25rem' }}
        >
          {data}
        </Heading>
      </Box>
      {icon && (
        <Box>
          <Box
            align="center"
            height="42px"
            justify="center"
            round="full"
            style={{ backgroundColor: colors[color] }}
            width="42px"
          >
            <Icon color="white" size="28px" />
          </Box>
        </Box>
      )}
    </Box>
  )
}

Pill.propTypes = propTypes

Pill.defaultProps = defaultProps

export default Pill
