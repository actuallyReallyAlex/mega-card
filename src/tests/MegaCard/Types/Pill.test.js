import React from 'react'
import Pill from '../../../MegaCard/Types/Pill'
import { shallow } from 'enzyme'

const defaultProps = {
  data: 'Test Data',
  title: 'Title',
  type: 'pill'
}

describe('<Pill />', () => {
  afterEach(() => {
    // defaultProps.handler.mockClear()
  })

  test('Should render the <Pill /> component.', () => {
    const component = shallow(<Pill {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
