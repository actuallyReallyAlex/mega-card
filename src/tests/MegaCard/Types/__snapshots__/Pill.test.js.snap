// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill /> Should render the <Pill /> component. 1`] = `ShallowWrapper {}`;
